### DISCUSSION TITLE
Monitor mode and single directory

### DISCUSSION THREAD

**Discussion by TommyPack** [abraunegg/onedrive#2972] [Q&A] [answered]
Hi thanks for this amazing project.
I started things off with the --monitor but then i recognized that i just need one specific folder on the linux PC and did the --single-directory /PATHtoDirectory --sync.
1. This all worked fine, but did i now switch from --monitor mode and the rest will never be synced now?
2. How can i now delete the files i don't need without them being deleted from my onedrive?
I could not find the specific command for deleting just on the local drive not whole onedrive anywhere in the --help.

Thank you in advance for your help.
Best,
Thomas

**Comment by abraunegg** [comment, marked as answer]
> Hi thanks for this amazing project. 

Thanks for your feedback - please consider sponsoring :)

> This all worked fine, but did i now switch from --monitor mode and the rest will never be synced now?

This depends ... did you enable a systemd service  to have the client running in the background? If you did then no .. you are still syncing *everything* and need to manually update the systemd service files with your desired configuration

> How can i now delete the files i don't need without them being deleted from my onedrive?

Firstly ensure that *all* application processes are stopped, all systemd service files are stopped. Ubuntu has a damn horrid default that it will enable a background systemd service even if you did not ask it do ... so please ensure that this is all taken care of.

Once this is stopped, reboot your system to ensure that there is no 'onedrive' process running *anywhere*: `sudo ps aufxw | grep onedrive` ... if that returns a process running 'onedrive' ... then you still have work to do.

Once nothing is running, just delete the files locally.

On the *next* restart / use of the application, ensure you use the `--resync` flag to ensure the local database is also cleaned up and now only contains all your `--single-directory` references.

Please also note, a bug was identified in v2.5.2 that was fixed post v2.5.2 that impacts `--single-directory` use .... v2.5.3 will include that fix.

**Reply by TommyPack** [reply]
> This depends ... did you enable a systemd service to have the client running in the background? If you did then no .. you are still syncing everything and need to manually update the systemd service files with your desired configuration

I really don't know tbh, I just wanted to have some of my files on the laboratory PC and work on them so there is not that much knowledge about systemd and stuff..

> Firstly ensure that all application processes are stopped, all systemd service files are stopped. Ubuntu has a damn horrid default that it will enable a background systemd service even if you did not ask it do ... so please ensure that this is all taken care of.

How would I do that ? 🤣

**Reply by abraunegg** [reply]
> How would I do that ?

1. Dont use Ubuntu .. use a better distribution.
2. If you must use Ubuntu - install the client the correct way if you use that platform: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md - this details how to take care of those erroneous systemd services
3. Please read the documentation on starting|stopping|enabling systemd services

**Reply by TommyPack** [reply]
Thank you sir,
if i want to check if something is running, I get this message:
```
admire@admire-Alienware-Aurora-R7:~$ sudo ps aufxw | grep onedrive
admire      5464  0.0  0.0   9564  2432 pts/0    SN+  12:19   0:00          \_ grep --color=auto onedrive
```
Does this mean something is running?
I tried to stop the service like the documentation stated.
BR
Tommy

**Reply by abraunegg** [reply]
Based on that output, there appears to be no 'onedrive' client running, only the 'grep' process
